coderd/database/migrations/000258_add_otp_reset_to_user.up.sql

```diff
@@ -0,0 +1,8 @@
+ALTER TABLE users ADD COLUMN hashed_one_time_passcode bytea;
+COMMENT ON COLUMN users.hashed_one_time_passcode IS 'A hash of the one-time-passcode given to the user.';
+
+ALTER TABLE users ADD COLUMN one_time_passcode_expires_at timestamp with time zone;
+COMMENT ON COLUMN users.one_time_passcode_expires_at IS 'The time when the one-time-passcode expires.';
+
+ALTER TABLE users ADD COLUMN must_reset_password bool NOT NULL DEFAULT false;
+COMMENT ON COLUMN users.must_reset_password IS 'Determines if the user should be forced to change their password.';
```

Comment:
Do we need this column? Could we instead say that a user must reset their password if `hashed_one_time_passcode IS NOT NULL`?

Reply:
By doing that you could force someone to change their password by putting someone's email in the "Forgot password?" form (when it exists). That is why it is a separate field.